src/System.Private.ServiceModel/tools/setupfiles/BuildWCFSelfHostedService.cmd

```diff
@@ -0,0 +1,49 @@
+@echo off
+setlocal
+
+if not defined VisualStudioVersion (
+    if defined VS140COMNTOOLS (
+        call "%VS140COMNTOOLS%\VsDevCmd.bat"
+        goto :EnvSet
+    )
+
+    if defined VS120COMNTOOLS (
+        call "%VS120COMNTOOLS%\VsDevCmd.bat"
+        goto :EnvSet
+    )
+
+    echo Error: %~nx0 requires Visual Studio 2013 or 2015.
+    echo        Please see https://github.com/dotnet/wcf/wiki/Developer-Guide for build instructions.
+    exit /b 1
+)
+
+:EnvSet
+
+:: Log build command line
+set _buildproj1=%~dp0..\SelfHostedWcfService\SelfHostedWCFService.sln
+set _buildlog=%~dp0..\..\..\..\msbuildSelfHostedWcfService.log
+set _buildprefix=echo
+set _buildpostfix=^> "%_buildlog%"
+
+call :build %*
+
+:: Build
+set _buildprefix=
+set _buildpostfix=
+call :build %*
+
+goto :AfterBuild
```

Comment:
I may not follow... but looks like the 4 lines above will never run and this script ends up just display the build command you plan to run.

Normally, I will do it this way... hopefully it will be easier to read

```
set cmd=msbuild "%_buildproj1%" /nologo /maxcpucount /verbosity:minimal /nodeReuse:false /fileloggerparameters:Verbosity=diag;LogFile="%_buildlog%";Append %*
echo %cmd%
%cmd%> "%_buildlog%"
```